client/layout/index.jsx

```diff
@@ -324,7 +325,10 @@ export default withCurrentRoute(
 		( state, { currentSection, currentRoute, currentQuery, secondary } ) => {
 			const sectionGroup = currentSection?.group ?? null;
 			const sectionName = currentSection?.name ?? null;
-			const siteId = getSelectedSiteId( state );
+			// Falls back to using the user's primary site if no site has been selected
+			// by the user yet
+			const currentSelectedSiteId = getSelectedSiteId( state );
+			const siteId = currentSelectedSiteId || getPrimarySiteId( state );
```

Comment:
We generally use the `getSelectedOrPrimarySiteId()` selector when we want to fall back to the primary site if no other one is currently selected.